op-guide/upgrade-FAQ.md

```diff
@@ -0,0 +1,200 @@
+
+```shell
+▶ curl "http://$IP:10080/schema/test/t" | python -m json.tool  # 这里用了 python 的格式化 json的工具，也可以不加，此处只是为了方便注释。
+{
+    "ShardRowIDBits": 0,
+    "auto_inc_id": 0,
+    "charset": "utf8",   # 这是 table 的 charset
+    "collate": "",
+    "cols": [			 # 从这里开始列举 column 的相关信息
+        {
+            ...
+            "id": 1,
+            "name": {
+                "L": "a",
+                "O": "a"   # column 的名字
+            },
+            "offset": 0,
+            "origin_default": null,
+            "state": 5,
+            "type": {
+                "Charset": "utf8",   # column a 的 charset
+                "Collate": "utf8_bin",
+                "Decimal": 0,
+                "Elems": null,
+                "Flag": 0,
+                "Flen": 10,
+                "Tp": 15
+            }
+        }
+    ],
+    ...
+}
+```
+
+### 问题2 : unsupported modify charset from utf8mb4 to utf8
+
+升级前：v2.1.1, v2.1.2
+
+```SQL
+tidb > create table t(a varchar(10)) charset=utf8;
+Query OK, 0 rows affected
+Time: 0.109s
+tidb > show create table t
++-------+-------------------------------------------------------+
+| Table | Create Table                                          |
++-------+-------------------------------------------------------+
+| t     | CREATE TABLE `t` (                                    |
+|       |   `a` varchar(10) DEFAULT NULL                        |
+|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
++-------+-------------------------------------------------------+
+```
+
+上面 `show create table` 只 show 出了 table 的 charset，但其实 column 的 charset 是 UTF8MB4，这可以用 http api 获取 schema 来确认。 这是一个 bug，即此处建表时 column 的 charset 应该要和 table 保持一致为 UTF8，这个问题在  v2.1.3 后已经修复。
+
+升级后：v2.1.3 以及之后
+
+```SQL
+tidb > show create table t
++-------+--------------------------------------------------------------------+
+| Table | Create Table                                                       |
++-------+--------------------------------------------------------------------+
+| t     | CREATE TABLE `t` (                                                 |
+|       |   `a` varchar(10) CHARSET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL |
+|       | ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin              |
++-------+--------------------------------------------------------------------+
+1 row in set
+Time: 0.007s
+tidb > alter table t change column a a varchar(20);
+ERROR 1105 (HY000): unsupported modify charset from utf8mb4 to utf8
+```
+
+因为 v2.1.3 之后支持修改 column 和 table 的 charset 了，所以这里推荐修改 table 的 charset 为 UTF8MB4。
+
+```SQL
+alter table t convert to character set utf8mb4;
+```
+
+也可以像问题 1 一样指定 column 的 charset，保持和 column 原来的 charset  UTF8MB4 一致即可。
+
+```SQL
+alter table t change column a a varchar(20) character set utf8mb4;
+```
+
+### 问题3 : ERROR 1366 (HY000): incorrect utf8 value f09f8c80(🌀) for column a
+
+TiDB 在 v2.1.1 以及之前，如果 charset 是 UTF8, 是没有对插入数据是 4-byte UTF8 unicode encoding 检查，在v2.1.2 以及之后，添加了检查。
```

Comment:
```suggestion
TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte 的插入数据进行 UTF8 Unicode encoding 检查。在v2.1.2 及之后版本中，添加了该检查。
```

Reply:
Please check whether this is what you mean. The original text is a little bit confusing.